core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/GlobalDefinitionCamelModelElement.java

```diff
@@ -30,7 +30,10 @@ public GlobalDefinitionCamelModelElement(AbstractCamelModelElement parent, Node
 	 */
 	@Override
 	protected void parseNode() {
+		if (getNodeTypeId().equalsIgnoreCase(BEAN_NODE)) {
```

Comment:
seems that we have some code related to some of the subclass inside this class.
I think that it is best to avoid this situation.
2 proposals:
- define an interface for GlobalDefinitionCamelModelElement, with 2 implementations inheriting from AbstractCamelModelElement and implementing this interface. One for "classical" GlobalDefinitionCamelModelElement and one for CamelBean
- define a "shouldParseNode()" method in AbstractCamelModelElement returinign true by default and override it in GlobalDefinitionCamelModelement to return false and in CmaleBean to return true. In AbstractCamelModelElement, the parseNode() { if shouldParseNode() {// first parse direct attributes
		parseAttributes();

		// now parse child elements
		parseChildren();

		// link child attributes like expressions to parent parameters
		linkChildrenToAttributes();}}

What do you think?
Other ideas?